cob_monitoring/src/battery_light_monitor.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/python
+#################################################################
+##\file
+#
+# \note
+#   Copyright (c) 2016 \n
+#   Fraunhofer Institute for Manufacturing Engineering
+#   and Automation (IPA) \n\n
+#
+#################################################################
+#
+# \note
+#   Project name: care-o-bot
+# \note
+#   ROS stack name: cob_command_tools
+# \note
+#   ROS package name: cob_monitoring
+#
+# \author
+#   Author: Benjamin Maidel
+# \author
+#   Supervised by:
+#
+# \date Date of creation: JAN 2015
+#
+# \brief
+#   Monitors the battery level and announces warnings and reminders to recharge.
+#
+#################################################################
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#     - Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer. \n
+#     - Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in the
+#       documentation and/or other materials provided with the distribution. \n
+#     - Neither the name of the Fraunhofer Institute for Manufacturing
+#       Engineering and Automation (IPA) nor the names of its
+#       contributors may be used to endorse or promote products derived from
+#       this software without specific prior written permission. \n
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License LGPL as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Lesser General Public License LGPL for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License LGPL along with this program.
+# If not, see <http://www.gnu.org/licenses/>.
+#
+#################################################################
+
+import rospy
+
+from cob_msgs.msg import *
+from cob_light.msg import *
+from std_msgs.msg import *
+from cob_light.srv import *
+
+
+class battery_light_monitor():
+  def __init__(self):
+    self.power_state = PowerState()
+    self.relative_remaining_capacity = 0.0
+    self.temperature = 0.0
+    self.is_chargeing = False
+    self.num_leds = rospy.get_param('num_leds', 58)
+
+    rospy.logdebug("waiting for service server")
+    rospy.wait_for_service('/light_torso/set_light')
+    rospy.logdebug("found service server")
+    try:
+      self.sproxy = rospy.ServiceProxy('/light_torso/set_light', SetLightMode)
```

Comment:
If the topic name is  parameterizable...it could be used to set other lights as well.....e.g. for r@w's with according wappers for the bms....